### PR TITLE
Update to use `sockets` instead of `numa_domains` with pika's resource partitioner

### PR DESCRIPTION
**Pull request by msimberg**
Fixes a breaking change introduced in pika 0.28.0.

I've taken the opportunity to fix a TODO that required a newer version of pika: check whether pika has been initialized before accessing the resource partitioner.

### CODE REVIEW

File: ci/docker/common.yaml [hunk 1]
```diff
@@ -46,3 +46,7 @@ packages:
     # Force git as non-buildable to allow deprecated versions in environments
     # https://github.com/spack/spack/pull/30040
     buildable: false
+  # TODO: Remove before merging
```
Comment (msimberg) [inline ci/docker/common.yaml:49]: TODO, should use spack commit that has pika 0.28.0 available.

**Comment by msimberg** [comment]
cscs-ci run

**Comment by msimberg** [comment]
This will keep failing in CI for now because https://github.com/spack/spack/pull/46062 is required to find Boost properly with pika 0.28.0. Once https://github.com/spack/spack/pull/46207 is merged I'll update the spack commit here.

**Comment by msimberg** [comment]
Note that this breakage is another great reason to move the MPI pool initialization into pika with #1125.

**Comment by msimberg** [comment]
Once I update the spack commit, it might make sense to make some CI configurations use older, fixed, versions of pika. Any objections or wishes for which pipelines should have older versions if any?

File: ci/docker/common.yaml (hunk 1 above)
Comment (msimberg) [reply ci/docker/common.yaml]: Updated.

**Comment by msimberg** [comment]
cscs-ci run

**Comment by msimberg** [comment]
cscs-ci run

**Review by rasolca — APPROVED** [review]

**Comment by msimberg** [comment]
cscs-ci run

File: ci/docker/debug-cuda-scalapack.yaml [hunk 2]
```diff
@@ -33,7 +33,11 @@ spack:
     mpich:
       require:
         - '+fortran'
+    fmt:
+      require:
+        - '@10'
```
Comment (msimberg) [inline ci/docker/debug-cuda-scalapack.yaml:38]: I've added a conflict upstream: https://github.com/spack/spack/pull/46280. The conflict mentions https://github.com/pika-org/pika/pull/1074.

Comment (msimberg) [reply ci/docker/debug-cuda-scalapack.yaml]: The conflict is merged and I've updated the spack sha here. I've removed the explicit version constraint on this PR again.

**Comment by msimberg** [comment]
cscs-ci run

**Comment by msimberg** [comment]
This is now ready for review. The disabled stdexec pipelines are tracked in #1184. The clang-based stdexec pipelines are still enabled, so we do still have some stdexec tests running even with the GCC ones disabled for now.

**Review by albestro — APPROVED** [review]

**Review by rasolca — APPROVED** [review]

**Pull request merged**